include/sys/util.h

```diff
@@ -770,6 +770,92 @@ u8_t u8_to_dec(char *buf, u8_t buflen, u8_t value);
  */
 #define UTIL_LISTIFY(LEN, F, ...) UTIL_EVAL(UTIL_REPEAT(LEN, F, __VA_ARGS__))
 
+/**
+ * @brief Get number of items on a list. It works with lists up to
+ * 20 items.
+ *
+ * Example:
+ *    \#define LIST led0, led1, led5
+ *    int leds = UTIL_LIST_LEN(LIST);
+ *    will assign 3 to leds.
+ */
+#define UTIL_LIST_LEN_(_0, _1, _2, _3, _4, _5, _6, _7, _8, _9, _10, \
+		     _11, _12, _13, _14, _15, _16, _17, _18, _19,   \
+		     len, ...) len
+
+#define UTIL_LIST_LEN(...) \
+	UTIL_LIST_LEN_(__VA_ARGS__,				   \
+		       20, 19, 18, 17, 16, 15, 14, 13, 12, 11, 10, \
+		       9, 8, 7, 6, 5, 4, 3, 2, 1, 0)
+
+/**
+ * @brief Invoke function/macro for each item on list. It works with
+ * lists with up to 20 items.
+ *
+ * @param fun	The function/macro to invoke, it takes one parameter,
+ *		that is currently processed list item.
+ *
+ * Example:
+ *    \#define GEN(itm) this_is_enum_ \#\# itm,
+ *    \#define LIST one, two, thre
+ *    enum {
+ *        UTIL_FOR_LIST(GEN, LIST)
+ *    };
+ *    will generate:
+ *    enum {
+ *        this_is_enum_one, this_is_enum_two, this_is_enum_three
+ *    };
+ *
+ * @note Calling with empty list generates empty string.
+ */
+#define UTIL_FOR_LIST(fun, ...) \
```

Comment:
Actually, this looks functionally identical to MACRO_MAP. Am I missing something? If not, why add this?

For example, this:

```
#define GEN(item) this_is_enum_ ## item,
#define LIST one, two, three
enum {
    MACRO_MAP(GEN, LIST)
};
```

produces the same results.

Reply:
Removed it.